pkg/apiserver/apiserver.go

```diff
@@ -112,6 +116,38 @@ const (
 	MaxTimeoutSecs = 600
 )
 
+func (g *APIGroupVersion) GetAPIPrefix() string {
+	slashlessRoot := strings.Trim(g.Root, "/")
+	if lastSlashIndex := strings.LastIndex(slashlessRoot, "/"); lastSlashIndex != -1 {
+		return slashlessRoot[:lastSlashIndex]
+	}
+
+	return slashlessRoot
+}
+
+func (g *APIGroupVersion) GetAPIGroup() string {
+	slashlessRoot := strings.Trim(g.Root, "/")
+	if lastSlashIndex := strings.LastIndex(slashlessRoot, "/"); lastSlashIndex != -1 {
+		return slashlessRoot[lastSlashIndex:]
+	}
+
+	return ""
+}
+
+func (g *APIGroupVersion) GetAPIVersion() string {
+	return g.Version
+}
+
+func (g *APIGroupVersion) GetAPIRequestInfoResolver() *APIRequestInfoResolver {
+	apiPrefix := g.GetAPIPrefix()
```

Comment:
If these methods aren't used anywhere but in newInstallsr, it would be better to have a method newAPIGroupInfoResolver that takes APIGroupVersion and creates a request info resolver.  Adding a bunch of methods on the type is confusing and people might use them.